addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1026,6 +1027,59 @@ options.registry.anchor = options.Class.extend({
     },
 });
 
+options.registry.CookiesBar = options.registry.SnippetPopup.extend({
+    xmlDependencies: (options.registry.SnippetPopup.prototype.xmlDependencies || []).concat(
+        ['/website/static/src/xml/website.cookies_bar.xml']
+    ),
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Change the cookies bar layout.
+     *
+     * @see this.selectClass for parameters
+     */
+    selectLayout: function (previewMode, widgetValue, params) {
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: function (ctx) {
+                websiteId = ctx['website_id'];
+            },
+        });
+
+        const $template = $(qweb.render(`website.cookies_bar.${widgetValue}`, {
+            websiteId: websiteId,
+        }));
+
+        const $content = this.$target.find('.s_popup_content');
+        const selectorsToKeep = [
+            '.o_cookies_bar_text_button',
+            '.o_cookies_bar_text_policy',
+            '.o_cookies_bar_text_title',
+            '.o_cookies_bar_text_primary',
+            '.o_cookies_bar_text_secondary',
+        ];
+
+        if (this.allChildNodes === undefined) {
+            this.allChildNodes = [];
+        }
+
+        for (const selector of selectorsToKeep) {
+            const $el = $content.find(selector);
+            if ($el.length && $el[0].childNodes.length) {
+                // save value before change, eg 'title' is not inside 'discrete' template
+                // but we want to preserve it in case of select another layout later
+                this.allChildNodes[selector] = $el[0].childNodes;
+            }
+            $template.find(selector).html(this.allChildNodes[selector]);
+        }
```

Comment:
I see you changed with the `<div and unique class> stuff` so you don't need to use the childNodes etc. Actually I am not sure what happens with `.html(childNodes)`, it is actually not allowed by jQuery documentation so I don't know if the nodes are recreated or not.

Can't the code be:
```javascript
        for (const selector of selectorsToKeep) {
            const $el = $content.find(selector);
            if ($el.length) {
                $template.find(selector).replaceWith($el);
            }
        }
```
No need of the tempory allChildNodes array right? + perform a move instead of a DOM duplication

Reply:
No, it's not that trivial.
For layout A B C, some elements are only related to B and C for instance, we then need to keep the A-related element inside the array, for when the user will get back to A.
Also, some elements required in both A B C could be missing if the user deleted the element. eg in layout A, an element was removed, we need to restore it when he moves to B or C.
It goes both ways really, there might be missing element or not needed element when switching from a layout to another.

Anyway, despite the fact it was working as it should on both browser, as it was not jQuery compliant, I refactored that part to not pass DOM element to `$.html()`.

Reply:
Ok, I thought there might be something like it but I was not sure anymore. Note: it might not seem important to use append/replace, but I guarantee you this could be. It would be normal for an option / an editor code to rely on DOM nodes to not be duplicated (because saved in a variable / event handlers attached / whatever) :wink: 

Anyways, if you changed everything, let's send this to the testing you want. India? Sbu? :wink:

Reply:
Task has already moved through the testing phase (sbu).
You can move forward with this one